File: problem_7/python/problem7.py
#!/usr/bin/python3

import math

def is_prime(divided):
  divisor = 3
  sqrt_divided = math.sqrt(divided)
  while divisor <= sqrt_divided:
    if divided % divisor == 0:
      return False

    divisor += 2
  return True

if __name__ == "__main__":
  prime_list = [2]
  count = 3

  while len(prime_list) < 10001:
    if is_prime(count) == True:
      prime_list.append(count)

    count += 2

  print(prime_list[-1])

File: problem_7/perl/problem7.pl
#!/usr/bin/perl
#
#
sub is_prime
{
  my ($divided) = @_;
  my $divisor = 3;
  my $sqrt_divided = sqrt($divided);

  while($divisor <= $sqrt_divided)
  {
    return 0 unless ($divided % $divisor);
    $divisor += 2;
  }

  1;
}


my $count = 1;
my $last;

for ( my $i = 3; $count < 10001; $i += 2)
{
  if (is_prime($i))
  {
    $count += 1;
    $last = $i;
  }
}

print $last;
